### PR TITLE
refactor: Lint Kotlin, C and C++ code

### PR DESCRIPTION
**Pull request by luanpotter**

### CODE REVIEW

File: .editorconfig
Comment (Gustl22) [inline .editorconfig:1]: As Note: if not planning to lint kotlin in the future, can remove the editorconfig then. The config doesn't hurt though, to provide restrictions if want to introduce Bluetooth Gamepads or sth on Android.

Comment (luanpotter) [reply .editorconfig:1]: I will remove for now, good catch. We can always re-add if we ever have other languages

File: .github/workflows/cicd.yml [hunk 1]
```diff
@@ -49,6 +49,17 @@ jobs:
...
       - run: markdownlint . -p .markdownlintignore -c .markdownlint.yaml
+
+  super-linter:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: super-linter/super-linter@v5
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          VALIDATE_ALL_CODEBASE: true
```
Comment (Gustl22) [inline .github/workflows/cicd.yml:60]: I think `true` only makes sense to validate the first time. But after that only the changed files are checked, which I think is sufficient if not pushing directly.

**Review by Gustl22 — APPROVED** [review]
LGTM

**Review by spydon — APPROVED** [review]

**Comment by Gustl22** [comment]
I missed doing a complete checkout in order to work with foreign repositories, also we can use the slim image...
Maybe want to apply these changes here, too: https://github.com/bluefireteam/audioplayers/pull/1624

**Review by renancaraujo — APPROVED** [review]

**Pull request merged**